backend/inspirehep/search/facets.py

```diff
@@ -225,7 +225,7 @@ def citation_summary():
         "citation_count_without_self_citations",
     ]
     filters = get_filters_without_excluded(hep_filters(), excluded_filters)
-    if request.values.get("exclude-self-citations", False):
+    if request.values.get("exclude-self-citations", False, type=bool):
```

Comment:
`bool("false") == True`. Why not simply
```suggestion
    if "exclude-self-citations" in request.values:
```
then it doesn't matter what you set it to: if it's present, it's interpreted as `True` whatever it's set to, otherwise as `False`.

Reply:
IMHO it makes code less readable.

Reply:
IDK what others thinks?

Reply:
It's not about readability but about behavior. I think it's surprising that `/facets?exclude-self-citations=` doesn't exclude self-citations, but `/facets?exclude-self-citations=false` does.

Reply:
but `/facets?exclude-self-citations=false` also won't exclude self-citations.

Reply:
No, you are right. Now any value in the parameter is True :/ So it's not what I wanted, I will go back to json.load then or I will go with your suggestion.

Reply:
:+1:

Reply:
nice idea, IMHO this is not how you standarize an api and it's more towards legacy; send a value and we'll guess what to do with it. ``true`` & ``false`` are universal and objective, they can be validated and documented easily with standard tools.